### PR TITLE
Plugins: Remove name() and description() from api

### PR DESCRIPTION
**Pull request by rjernst**
In 2.0 we added plugin descriptors which require defining a name and
description for the plugin. However, we still have name() and
description() which must be overriden from the Plugin class. This still
exists for classpath plugins. But classpath plugins are mainly for
tests, and even then, referring to classpath plugins with their class is
a better idea. This change removes name() and description(), replacing
the name for classpath plugins with the full class name.

### CODE REVIEW

File: plugins/discovery-ec2/src/main/java/org/elasticsearch/plugin/discovery/ec2/Ec2DiscoveryPlugin.java [hunk 1]
```diff
@@ -67,23 +67,6 @@ public Void run() {
         });
     }
 
-    private final Settings settings;
-    protected final ESLogger logger = Loggers.getLogger(Ec2DiscoveryPlugin.class);
-
-    public Ec2DiscoveryPlugin(Settings settings) {
```
Comment (dadoonet) [inline plugins/discovery-ec2/src/main/java/org/elasticsearch/plugin/discovery/ec2/Ec2DiscoveryPlugin.java]: Indeed. Was probably a left over. :)

**Comment by dadoonet** [comment]
LGTM

**Comment by s1monw** [comment]
LGTM - nice cleanup these plugin API might actually be nice and simple at some point

**Comment by s1monw** [comment]
I just ran into some javadocs that reference these methods as an example in `AnalysisModule` you might wanna remove that too

**Comment by s1monw** [comment]
LGTM thx ryan

**Pull request merged**